101-cosmosdb-create-account/azuredeploy.json

```diff
@@ -0,0 +1,48 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "name": {
+      "type": "string",
+      "defaultValue": "[uniqueString('cosmosdb', resourceGroup().id)]",
+      "metadata": {
+        "description": "The name of your CosmosDB account"
+      }
+    },
+    "location": {
+      "type": "string",
+      "defaultValue": "[resourceGroup().location]",
+      "metadata": {
+        "description": "Location for your CosmosDB account"
+      }
+    },
+    "tier": {
+      "type": "string",
+      "defaultValue": "Standard",
+      "metadata": {
+        "description": "CosmosDB account tier"
+      }
+    }
+  },
+  "resources": [
+    {
+      "name": "[parameters('name')]",
+      "type": "Microsoft.DocumentDB/databaseAccounts",
+      "apiVersion": "2015-04-08",
```

Comment:
update apiVersion

Reply:
Hello @bmoore-msft 
As per the ARM schema definition for CosmosDB there's currently only 1 available API version for this resource provider which is "2015-04-08"
[https://docs.microsoft.com/en-us/azure/templates/microsoft.documentdb/allversions](url)
[https://docs.microsoft.com/en-us/azure/templates/microsoft.documentdb/2015-04-08/databaseaccounts](url)

Am I wrong ?

Olivier

Reply:
the /providers api will tell you the available versions - you can use resource explorer in the portal or az providers list -o json in the cli to get the latest

Reply:
Ok by running the cmd line under Azure CLI I got the following output : 

`"apiProfiles": [
          {
            "apiVersion": "2015-04-08",
            "profileVersion": "2018-06-01-profile"
          }
        ],
        "apiVersions": [
          "2016-03-31",
          "2016-03-19",
          "2015-11-06",
          "2015-04-08",
          "2014-04-01"
        ],`

Should I use "2016-03-31" then ?
So question is about the apiVersions tab decorelated of the apiProfiles[0].apiVersion value ??
Other question is about how to well define the ARM template related to a version which is not documented at all on the Azure documentation (I may miss required field as well as not understanding the behavior of that template once deployed) ?
Regarding [https://docs.microsoft.com/en-us/azure/templates/microsoft.documentdb/2015-04-08/databaseaccounts](url) I don't have any other apiVersion from the combo box

Thanks

Reply:
> There was a breaking change in a new version of node/npm - update your .travis.yml file to pick up that change or open a new PR after pulling from master.

Could you clarify please ? Not sure to understand the link with my PR.
Thanks

Reply:
hello @bmoore-msft do you have any update please ? thanks
Olivier

Reply:
re: apiVersion - use the latest reported by the providers /api... docs can lag if the RP has not updated  recently.

re: the build break, look at the .travis.yml file in your branch - make sure it matches master.  If not you need to update it in your branch for CI to run.

Reply:
Hello @bmoore-msft 
It should be fixed now : apiVersion has been updated with the last available version.
Travis file has been updated with the master branch content.
Olivier

Reply:
@bmoore-msft 
It seems the `resourceGroup().location` is causing the issue but -correct me if I'm wrong- this is something recommended by you right here : [https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md](url)
Any suggestion please ?

Reply:
@bmoore-msft do you have any comment please ?
As defined per your guidelines : 

> Templates must have a parameter named location for the primary location of resources.
The default value of this parameter must be [resourceGroup().location]

But this is what is causing the issue right here.
BTW it was working without any error before the release pipeline update, at the time of the creation of the PR.

Thanks

Reply:
BTW are you trying to execute `az deployment validate` command ? this one fail but validating the template through `az group deployment validate` works perfectly